File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Arrays;
import java.util.ArrayList;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Arrays.sort(participant);
        Arrays.sort(completion);
        for(int i = 0; i<completion.length ; i++){
            if(!completion[i].equals(participant[i])){
                answer = participant[i];
                return answer;
                }
            }
        //모든 요소가 매치된 경우 participant의 마지막 요소가 완주못한 사람
        answer = participant[participant.length - 1];
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    players = dict()
    complete = dict()
    for i in participant:
        players[i] = 0
        complete[i] = 0
    #print(players, complete)
    
    for i in participant:
        players[i] += 1
    #print(players)
    for i in completion:
        complete[i] += 1
    #print(complete)
    for i in players:
        if players[i] != complete[i]:
            answer += i
            break
    
    return answer
